packages/core/src/queries/tenant-usage/types.ts

```diff
@@ -1,67 +1,150 @@
+  booleanFeatureKeyGuard.safeParse(key).success;
+
+/**
+ * Numeric limit keys - numeric usage limits
+ *
+ * @remarks
+ * These represent countable limits that can be queried for current usage.
+ *
+ * @example
+ * - applicationsLimit
+ * - hooksLimit
+ * - organizationsLimit
+ */
+export type NumericLimitKey = {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  [K in keyof AllLimitMap]: AllLimitMap[K] extends number | null ? K : never;
+}[keyof AllLimitMap];
+
+const numericLimitKeyGuard = z.enum([
+  'applicationsLimit',
+  'thirdPartyApplicationsLimit',
+  'machineToMachineLimit',
+  'userRolesLimit',
+  'machineToMachineRolesLimit',
+  'scopesPerRoleLimit',
+  'scopesPerResourceLimit',
+  'hooksLimit',
+  'resourcesLimit',
+  'enterpriseSsoLimit',
+  'organizationsLimit',
+  'samlApplicationsLimit',
+  'socialConnectorsLimit',
+  'tenantMembersLimit',
+]) satisfies ToZodEnum<NumericLimitKey>;
+
+export const isNumericLimitKey = (key: AllLimitKey): key is NumericLimitKey =>
+  numericLimitKeyGuard.safeParse(key).success;
+/**
+ * Self-computed usage keys - usage data queried from local tenant database
+ *
+ * @remarks
+ * These usage values are calculated from the tenant's own database.
+ *
+ * **Why only numeric limits:**
+ * During quota checks, boolean features (feature availability) are checked directly
+ * against the plan quota - `true` means allowed, `false` means not allowed.
+ * Only consumable/numeric limits require actual usage calculation by querying
+ * the database to compare current usage against the limit.
+ *
+ * **Excluded keys:**
+ * Not all numeric limits are self-computed. The following require external data sources:
+ * - `tenantMembersLimit`: queried from Cloud API
+ * - `socialConnectorsLimit`: queried from Connector Library
+ */
+export type SelfComputedUsageKey = Exclude<
+  NumericLimitKey,
+  'tenantMembersLimit' | 'socialConnectorsLimit'
 >;
 
-export const tenantUsageGuard = z.object({
-  applicationsLimit: z.number(),
-  thirdPartyApplicationsLimit: z.number(),
-  scopesPerResourceLimit: z.number(),
-  userRolesLimit: z.number(),
-  machineToMachineRolesLimit: z.number(),
-  scopesPerRoleLimit: z.number(),
-  hooksLimit: z.number(),
-  customJwtEnabled: z.boolean(),
-  bringYourUiEnabled: z.boolean(),
-  collectUserProfileEnabled: z.boolean(),
-  /** Add-on quotas start */
-  machineToMachineLimit: z.number(),
-  resourcesLimit: z.number(),
-  enterpriseSsoLimit: z.number(),
-  mfaEnabled: z.boolean(),
-  organizationsLimit: z.number(),
-  securityFeaturesEnabled: z.boolean(),
-  /** Enterprise only add-on quotas */
-  idpInitiatedSsoEnabled: z.boolean(),
-  samlApplicationsLimit: z.number(),
-  /** Add-on quotas end */
-}) satisfies ToZodObject<TenantUsage>;
+/**
+ * Self-computed usage keys runtime guard
+ *
+ * @remarks
+ * This guard validates all usage keys that can be computed from local database.
+ *
+ */
+const selfComputedUsageKeyGuard = z.enum([
```

Comment:
Is it possible to simply exclude `socialConnectors` and `tenantMembersLimit` from the `numericLimitKeyGuard` instead of defining a full list for `selfComputedUsageKeyGuard`?

Reply:
This type guard is unused, removed.